source/codegen/common_helpers.py

```diff
@@ -197,11 +197,12 @@ def is_unsupported_size_mechanism(parameter: dict) -> bool:
 #                     elements filled in to the array as long as that is smaller than the passed-in value. (if it is
 #                     larger, the underlying call will return an error)
 #                     Should only be used for output arrays.
+# - two-dimension: The array is two dimensional
```

Comment:
Can you put a little more information here. Basically that the size param is actually a list of sizes and the actual mechanism is just to validate that the 1-d input data is the same size as the sum of the list of sizes.